### ISSUE TITLE
join() fails when input array contains more than one variable reference

### ISSUE THREAD

**Issue by seh** [mikefarah/yq#1048]
# Description

If you create an array containing more than one variable reference and pipe it to `join`, _yq_ panics. Even with only one variable reference, _yq_ panics as well, depending on where in the array that reference occurs.

**Version of _yq_:** 4.16.2
**Operating system:** macOS
**Installed via:** HomeBrew

# Reproduction

```shell
yq eval \
   '"abc" as $first |
    "def" as $second |
    [$first, $second] | join(".")' \
   --null-input
```
# Actual Behavior

<details>
<summary>Go panic output</summary>

```
panic: interface conversion: interface {} is nil, not *yqlib.CandidateNode

goroutine 1 [running]:
github.com/mikefarah/yq/v4/pkg/yqlib.collectOperator(0xc000109440, {0xc00023f650, 0xc00023fbf0, 0x0}, 0xc0002da1b0)
        github.com/mikefarah/yq/v4/pkg/yqlib/operator_collect.go:65 +0x766
github.com/mikefarah/yq/v4/pkg/yqlib.(*dataTreeNavigator).GetMatchingNodes(0x13209ac, {0xc00023f650, 0xc00023fbf0, 0x10}, 0xc0002da1b0)
        github.com/mikefarah/yq/v4/pkg/yqlib/data_tree_navigator.go:37 +0x1c3
github.com/mikefarah/yq/v4/pkg/yqlib.pipeOperator(0xc000109440, {0xc00023f650, 0xc00023fbf0, 0x0}, 0xc0002da1f8)
        github.com/mikefarah/yq/v4/pkg/yqlib/operator_pipe.go:9 +0x3d
github.com/mikefarah/yq/v4/pkg/yqlib.(*dataTreeNavigator).GetMatchingNodes(0x13209ac, {0xc00023f650, 0xc00023fbf0, 0x10}, 0xc0002da1f8)
        github.com/mikefarah/yq/v4/pkg/yqlib/data_tree_navigator.go:37 +0x1c3
github.com/mikefarah/yq/v4/pkg/yqlib.pipeOperator(0xc000109440, {0xc00023f650, 0xc00023fbf0, 0x0}, 0xc0002da210)
        github.com/mikefarah/yq/v4/pkg/yqlib/operator_pipe.go:13 +0x65
github.com/mikefarah/yq/v4/pkg/yqlib.(*dataTreeNavigator).GetMatchingNodes(0x13209ac, {0xc00023f650, 0xc00023fbf0, 0x10}, 0xc0002da210)
        github.com/mikefarah/yq/v4/pkg/yqlib/data_tree_navigator.go:37 +0x1c3
github.com/mikefarah/yq/v4/pkg/yqlib.pipeOperator(0xc000109440, {0xc00023f650, 0x0, 0x0}, 0xc0002da228)
        github.com/mikefarah/yq/v4/pkg/yqlib/operator_pipe.go:13 +0x65
github.com/mikefarah/yq/v4/pkg/yqlib.(*dataTreeNavigator).GetMatchingNodes(0xc0002bfae0, {0xc00023f650, 0x0, 0xa0}, 0xc0002da228)
        github.com/mikefarah/yq/v4/pkg/yqlib/data_tree_navigator.go:37 +0x1c3
github.com/mikefarah/yq/v4/pkg/yqlib.(*streamEvaluator).EvaluateNew(0xc0002ebd88, {0x7ffeefbff188, 0xc000010018}, {0x139ac50, 0xc000125e30}, {0x0, 0x0})
        github.com/mikefarah/yq/v4/pkg/yqlib/stream_evaluator.go:47 +0x284
github.com/mikefarah/yq/v4/cmd.evaluateSequence(0xc000129180, {0xc000078500, 0x2, 0x2})
        github.com/mikefarah/yq/v4/cmd/evalute_sequence_command.go:123 +0x7af
github.com/spf13/cobra.(*Command).execute(0xc000129180, {0xc0000784e0, 0x2, 0x2})
        github.com/spf13/cobra@v1.3.0/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc000128f00)
        github.com/spf13/cobra@v1.3.0/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.3.0/command.go:902
main.main()
        github.com/mikefarah/yq/v4/yq.go:11 +0x1e
```
</details>

# Expected behavior

```yaml
abc.def
```

# Additional Context

Experimentation shows these similar input forms cause `join` to fail similarly:
- `["xyz", $first, "pdq"]`  
  That's only one variable.
- `[$first, "pdq"]`  
  Again, only one variable.

These similar input forms work as intended:
- `["pdq", $first]`
- `["xyz", "pdq", $first]`

**Comment by mikefarah** [comment]
Thanks @seh - I've been able to reproduce this. Will fix in the next release.

**Comment by mikefarah** [comment]
Doh git autoclosed this because of my commit message - regardless, fixed in 4.17.2
